### PR TITLE
feat(#684): add DAST with ZAP

### LINKED ISSUE #684: Feature request: Add DAST (ZAP) via github action
In order to pass OpenSSF criteria we need a github action that spins up a docker container and then uses ZAP to scan the app for any vulnerabilities

### PR DESCRIPTION
**Pull request by bendehaan**
### What kind of changes does this PR include?

-   [ ] Fixes or refactors
-   [ ] A new challenge
-   [X] Additional documentation
-   [X] Something else

#### Description

This PR adds a workflow to run the baseline scan against WrongSecrets running on Kubernetes. It will open and update a GitHub issue with potential problems.

### Relations

Closes #684

### References

Implemented the action according to [OWASP ZAP baseline scan action documentation](https://github.com/marketplace/actions/owasp-zap-baseline-scan)

### Checklist:

-   [x] All the contributions made are solely the work of me and my co-authors
-   [x] I tested the changes in this PR (if applicable)
-   [ ] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [x] The PR passes pre-commit hooks and automated tests

### CODE REVIEW

File: .github/workflows/minikube-zap-test.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# Controls when the workflow will run
+on:
+  pull_request:
+    branches: [master]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  issues: write
```
Comment (commjoen) [inline .github/workflows/minikube-zap-test.yml:13]: ???? why write?

Comment (commjoen) [reply .github/workflows/minikube-zap-test.yml:13]: Thanks! Can we instead just file the results as part of the job?

File: .github/workflows/minikube-zap-test.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Setup application
+        run: |
+          ./k8s-vault-minkube-start.sh
+      - name: ZAP Scan
+        uses: zaproxy/action-baseline@v0.7.0
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          issue_title: ZAP DAST scan findings
+          docker_name: "owasp/zap2docker-stable"
+          target: "http://localhost:8080"
```
Comment (commjoen) [inline .github/workflows/minikube-zap-test.yml:42]: Is it possible to add/maintain a suppression file :-)?

Comment (bendehaan) [reply .github/workflows/minikube-zap-test.yml]: Yes, adding it

File: .github/workflows/minikube-zap-test.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          minikube-version: 1.29.0
+          driver: docker
+          kubernetes-version: v1.25.6
+      - name: Setup helm
+        uses: azure/setup-helm@v3.5
+        id: install
+      - name: Setup application
+        run: |
+          ./k8s-vault-minkube-start.sh
```
[the remark this answers is not in the document]
Comment (bendehaan) [reply .github/workflows/minikube-zap-test.yml]: Alternatively run spring directly?

File: .github/workflows/minikube-zap-test.yml (hunk 1 above)
Comment (bendehaan) [reply .github/workflows/minikube-zap-test.yml]: Done

**Review by commjoen — APPROVED** [review]
Thanks! Lgtm! Can you please file an issue for the remaining findings :-)?

**Comment by bendehaan** [comment]
> Thanks! Lgtm! Can you please file an issue for the remaining findings :-)?

Will do!

**Pull request merged**